### PR TITLE
Investigating

### PR DESCRIPTION
**Pull request by jrobinAV**

### CODE REVIEW

File: .github/workflows/pr-tests-no-pytest.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install "git+https://git@github.com/Avaiga/taipy.git@fix/sqlrepo-integration-test" statsmodels Flask-Testing scikit-learn
+        timeout-minutes: 30
+
+      - name: Pytest test
```
Comment (jrobinAV) [inline .github/workflows/pr-tests-no-pytest.yml]:
```suggestion
      - name: Run test
```

File: Pipfile [hunk 2]
```diff
@@ -4,12 +4,10 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-taipy = {ref = "develop", git = "https://github.com/avaiga/taipy.git"}
+taipy = {ref = "fix/sqlrepo-integration-test", git = "https://github.com/avaiga/taipy.git"}
```
Comment (jrobinAV) [inline Pipfile]: This should be reverted before merging

File: tests/test_churn_classification/test_churn_classification.py [hunk 3]
```diff
@@ -36,13 +36,16 @@ def test_development_sql_repo(self, init_sql_repo):
...
 
+    @pytest.mark.churn_fs
     def test_standalone_fs_repo(self):
-        Config.configure_job_executions(mode=JobConfig._STANDALONE_MODE, max_nb_of_workers=2)
+        Config.configure_job_executions(mode=JobConfig._STANDALONE_MODE, max_nb_of_workers=4)
         self.waiting_jobs_to_complete = True
         self.__run()
 
+    @pytest.mark.churn_sql
+    @pytest.mark.skip("This test is skipped because it is not working as expected.")
```
Comment (jrobinAV) [inline tests/test_churn_classification/test_churn_classification.py]: Should check why it is skipped.

File: .github/workflows/pr-tests-no-pytest.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install "git+https://git@github.com/Avaiga/taipy.git@fix/sqlrepo-integration-test" statsmodels Flask-Testing scikit-learn
```
Comment (toan-quach) [inline .github/workflows/pr-tests-no-pytest.yml]: just put a comment here as a reminder to change this back to develop branch

File: .github/workflows/pr-tests.yml [hunk 5]
```diff
@@ -39,47 +20,25 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
-        run: pip install "git+https://git@github.com/Avaiga/taipy.git@develop" statsmodels Flask-Testing pytest scikit-learn mongomock
+        run: pip install "git+https://git@github.com/Avaiga/taipy.git@fix/sqlrepo-integration-test" statsmodels Flask-Testing pytest scikit-learn
```
Comment (toan-quach) [inline .github/workflows/pr-tests.yml]: just put a comment here as a reminder to change this back to develop branch

File: tests/test_churn_classification/algorithms.py [hunk 6]
```diff
@@ -165,12 +155,8 @@ def compute_roc(probabilities, test_dataset, partitions=100):
 
     roc_np = roc.reshape(-1, 2)
     roc_data = pd.DataFrame({"False positive rate": roc_np[:, 0], "True positive rate": roc_np[:, 1]})
-    # print("     Calculation done")
-    # print("     Scoring...")
-
     score_auc = roc_auc_score(y_test, probabilities)
-    # print("     Scoring done\n")
-
+    # print(f"{sys._getframe().f_code.co_name} - {time.time() - start}")
```
Comment (toan-quach) [inline tests/test_churn_classification/algorithms.py]: consider removing these comments

File: tests/test_churn_classification/algorithms.py [hunk 7]
```diff
@@ -146,15 +136,15 @@ def predict_baseline(test_dataset: pd.DataFrame, trained_model: LogisticRegressi
     Returns:
         forecast (pd.DataFrame): the forecasted dataset
     """
-    # print("     Forecasting the test dataset...")
+    start = time.time()
     X, y = test_dataset.iloc[:, :-1], test_dataset.iloc[:, -1]
     predictions = trained_model.predict_proba(X)[:, 1]
-    print("     Forecasting done!")
+    # print(f"{sys._getframe().f_code.co_name} - {time.time() - start}")
```
Comment (toan-quach) [inline tests/test_churn_classification/algorithms.py]: consider removing these comments

File: tests/test_churn_classification/algorithms.py [hunk 8]
```diff
@@ -130,11 +121,10 @@ def forecast(test_dataset: pd.DataFrame, trained_model: RandomForestClassifier):
     Returns:
         forecast (pd.DataFrame): the forecasted dataset
     """
-    # print("     Forecasting the test dataset...")
+    start = time.time()
     X, y = test_dataset.iloc[:, :-1], test_dataset.iloc[:, -1]
-    # predictions = trained_model.predict(X)
     predictions = trained_model.predict_proba(X)[:, 1]
-    # print("     Forecasting done!")
+    # print(f"{sys._getframe().f_code.co_name} - {time.time() - start}")
```
Comment (toan-quach) [inline tests/test_churn_classification/algorithms.py]: consider removing these comments

File: tests/test_churn_classification/algorithms.py [hunk 9]
```diff
@@ -112,13 +104,12 @@ def train(train_dataset: pd.DataFrame):
...
-    # print("     Training the model...\n")
+    start = time.time()
     X, y = train_dataset.iloc[:, :-1], train_dataset.iloc[:, -1]
-    model_fitted = RandomForestClassifier().fit(X, y)
-    # print("\n    ",model_fitted," is trained!")
-
+    model_fitted = RandomForestClassifier(n_estimators=50).fit(X, y)
     importance_dict = {"Features": X.columns, "Importance": model_fitted.feature_importances_}
     importance = pd.DataFrame(importance_dict).sort_values(by="Importance", ascending=True)
+    # print(f"{sys._getframe().f_code.co_name} - {time.time() - start}")
```
Comment (toan-quach) [inline tests/test_churn_classification/algorithms.py]: consider removing these comments

**Review by toan-quach — DISMISSED** [review]

File: tests/test_churn_classification/test_churn_classification.py (hunk 3 above)
Comment (joaoandre-avaiga) [reply tests/test_churn_classification/test_churn_classification.py]: This is intermitent, because the iteraction of the standalone mode with SQL.

**Comment by jrobinAV** [comment]
I merged this PR while the tests were failing. This must be resolved on a fresh new branch if needed.

**Pull request merged**